cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/management-credential-step/ManagementCredentials.tsx

```diff
@@ -153,6 +170,61 @@ function ManagementCredentials(props: Partial<StepProps>) {
         }
     };
 
+    useEffect(() => {
+        if (awsState[STORE_SECTION_FORM].REGION) {
+            retrieveOsImages(awsState[STORE_SECTION_FORM].REGION);
+        }
+    }, [awsState[STORE_SECTION_FORM].REGION]);
+
+    function retrieveOsImages(region: string) {
+        try {
+            setOsImages([]);
+            if (region !== '') {
+                AwsService.getAwsosImages(region).then((data) => {
+                    setOsImages(data);
+                    setErrorMessage((errorMessage) => {
+                        const copy = { ...errorMessage };
+                        delete copy['OS_IMAGE'];
+                        return copy;
+                    });
+                });
+            }
+        } catch (e) {
+            setErrorMessage({
+                ...errorMessage,
+                OS_IMAGE: e,
+            });
+        }
+    }
+
+    function showErrorInfo() {
+        if (connectionStatus === CONNECTION_STATUS.CONNECTED && JSON.stringify(errorMessage) !== '{}') {
+            return (
+                <div>
+                    <div className="error-text">Error Occurs</div>
```

Comment:
Nitpick: would change error text to read "Error Occurred"

Reply:
Yeah right. My typo. Correct it right away